### PR TITLE
problem: (fix #2034) initial HW wallet implementation is not integrated in client-*

### LINKED ISSUE #2034: Problem: initial HW wallet implementation is not integrated in client-*
currently, it only supports staking related operations (nodejoin,unbond,withdraw,unjail), but can already be integrated

### PR DESCRIPTION
**Pull request by linfeng-crypto**
solution:
* add ledger support in client-cli
* add staking address in devnet genesis.json and fix app hash in Makefile
* add libudev-dev pakcage and remove libzmq3-dev in the drone dockerfile

### CODE REVIEW

File: client-cli/src/command/address_command.rs [hunk 1]
```diff
@@ -146,7 +146,10 @@ impl AddressCommand {
         let wallet_kind = wallet_client.get_wallet_kind(name, &enckey)?;
         if wallet_kind == WalletKind::HW {
             let hw_kind = ask_hardware_kind(None)?;
-            wallet_client.update_hw_service(hw_kind);
+            wallet_client.update_hw_service(hw_kind).map_err(|e| {
+                error("maybe need sudo user");
```
Comment (tomtau) [inline client-cli/src/command/address_command.rs:150]: maybe better to point to this link: https://support.ledger.com/hc/en-us/articles/115005165269-Fix-connection-issues ?

Comment (tomtau) [reply client-cli/src/command/address_command.rs:150]: `sudo` is only needed for updating the udev rules

File: ledger-crypto-rs/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+[dev-dependencies]
+futures = "0.3.5"
+matches = "0.1.8"
+serial_test = "0.4.0"
+env_logger = "0.7.1"
+secp256k1 = "0.17.2"
+blake3 = "0.3.5"
+tokio = { version = "0.2", features = ["full"] }
+tokio-test = "0.2.1"
```
Comment (tomtau) [inline ledger-crypto-rs/Cargo.toml:42]: are all these dependencies necessary ? for `secp256k1`, could it use the same version as the rest of the repository?

Comment (linfeng-crypto) [reply ledger-crypto-rs/Cargo.toml:42]: I'll remove them, because I removed the integration tests in ledger-crypto-rs,  no need anymore.

**Comment by tomtau** [comment]
and Travis: https://github.com/crypto-com/chain/blob/master/.travis.yml#L37

File: client-core/src/service/ledger_hw_key_service.rs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    pub app: Arc<CryptoApp>,
+    /// confirmation on ledger or not
+    pub require_confirmation: bool,
+}
+
+impl std::fmt::Debug for LedgerService {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("LedgerService")
+            .field("app", &"CryptoApp")
+            .field("require_confirmation", &self.require_confirmation)
```
Comment (leejw51crypto) [inline client-core/src/service/ledger_hw_key_service.rs:27]: how about RequireConfirmation
cammel case

File: integration-tests/bot/client_cli.py [hunk 4]
```diff
@@ -274,9 +274,10 @@ def state(self, staking_address):
...
     def sync(
             self,
-            force=False,
-            disable_address_recovery=False,
+            force=True,
+            enable_address_recovery=False,
             enable_fast_forward=True,
+            disable_light_client=True,
             batch_size=20,
             block_height_ensure=50,
```
Comment (tomtau) [inline integration-tests/bot/client_cli.py:282]: why are these changes needed -- e.g. why no "disable_address_recovery"?

File: docker/config/devnet/tendermint/genesis.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-    "app_hash": "9B3263E4412E90902E7ED634399062A8DEDF45E89C87919E1474FD23EF77BB2F",
+    "app_hash": "D3218ECBF9D1F90656C624246ECDFB30CE01386F5BD5C6AC423766FEFC0DEB62",
```
Comment (tomtau) [inline docker/config/devnet/tendermint/genesis.json:2]: is this needed?
why not to update https://github.com/crypto-com/chain/blob/master/docker/config/devnet/dev-conf.json with it?

**Comment by tomtau** [comment]
@linfeng-crypto it looks like related to https://github.com/crypto-com/chain/issues/2093
```
pytests/test_client_cli.py .....FFFFFF                                   [ 78%]
1176	pytests/test_pending_tx.py .                                             [ 85%]
1177	pytests/test_wallet_offline.py .                                         [ 92%]
1178	pytests/test_watch_only_wallet.py .                                      [100%]
1179	
1180	=================================== FAILURES ===================================
1181	___________________ test_withdraw_all_unbonded_from_genesis ____________________
1182	
1183	    @pytest.mark.zerofee
1184	    def test_withdraw_all_unbonded_from_genesis():
1185	>       withdraw_transactions(wallet_1, staking_address_wallet_1)
1186	
1187	pytests/test_client_cli.py:196: 
1188	_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
1189	pytests/test_client_cli.py:154: in withdraw_transactions
1190	    wallet.sync()
1191	bot/client_cli.py:303: in sync
1192	    text = run(cmd, args)
```

**Comment by tomtau** [comment]
bors retry

**Comment by linfeng-crypto** [comment]
> #2093

yes, I got the same error and trying to fix it.

**Comment by tomtau** [comment]
bors try

**Comment by tomtau** [comment]
bors retry
